pages/booking/list.tsx

```diff
@@ -1,8 +1,118 @@
 // TODO restrict this only for users
+import { styled } from '@mui/material/styles';
+import { useState } from 'react'
+
+import Alert from '@mui/material/Alert'
+import Card from '@mui/material/Card';
+import CardActions from '@mui/material/CardActions';
+import CardContent from '@mui/material/CardContent';
+import Collapse from '@mui/material/Collapse';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import IconButton, { IconButtonProps } from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import useSWR from 'swr'
+
+interface ExpandMoreProps extends IconButtonProps {
+  expand: boolean;
+}
+
+const ExpandMore = styled((props: ExpandMoreProps) => {
+  const { expand, ...other } = props;
+  return <IconButton {...other} />;
+})(({ theme, expand }) => ({
+  transform: !expand ? 'rotate(0deg)' : 'rotate(180deg)',
+  marginLeft: 'auto',
+  transition: theme.transitions.create('transform', {
+    duration: theme.transitions.duration.shortest,
+  })
+}));
+
 function BookingList() {
-  return <div>
-    List of all bookings
-  </div>
+  const [message, setMessage] = useState(null)
+  const [expanded, setExpanded] = useState(false)
+  const fetcher = () => fetch('/api/booking/list').then(res => res.json())
+  const { data, error } = useSWR('/api/booking/list', fetcher)
+
+  if (error) {
+    setMessage({ 
+      severity: 'error',
+      text: "Failed to load"
+    })
+  }
+
+  fetch('/api/booking/list')
```

Comment:
shouldn't we use server side props?

Reply:
should I use the "getStaticProps" async function?